ote/src/clj/ote/services/pre_notices/attachments.clj

```diff
@@ -11,24 +11,44 @@
             [ote.db.transit :as transit]
             [ote.db.modification :as modification]
             [ote.components.http :as http]
-            [ote.nap.users :as users]))
+            [ote.nap.users :as users])
+  (:import (java.nio.file Files)))
+
+(def allowed-mime-types ["application/pdf" "image/jpeg" "image/png"])
 
 (defn- generate-file-key [id filename]
   (str id "_" filename))
 
+(defn validate-file [{:keys [tempfile]}]
+  (let [path (.toPath tempfile)
+        content-mime (Files/probeContentType path)]
+    (when-not (some #(= % content-mime) allowed-mime-types)
```

Comment:
If you changed `allowed-mime-types` from a vector to a set, you could simply check `(allowed-mime-types content-mime)`

Reply:
We should check (by name, I guess) on the frontend also so we don't upload something that is then rejected.

Reply:
Yes, true that. And a naive file-extension based validation could be a good choise for the front-end.